### PR TITLE
Get all user targets.

### PR DESCRIPTION
**Pull request by amaury901130**
### Description

- Get all target for the current user.

- No create more than 10 targets per user.

### CODE REVIEW

File: spec/requests/api/v1/targets/create_spec.rb [hunk 1]
```diff
@@ -45,4 +45,20 @@
       expect(json[:topic][:id]).to eq(topic_id)
     end
   end
+
+  context 'to many targets' do
```
Comment (gonzaM) [inline spec/requests/api/v1/targets/create_spec.rb]: Start contexts with `when` or `with` http://www.betterspecs.org/#contexts

File: spec/requests/api/v1/targets/create_spec.rb [hunk 2]
```diff
@@ -45,4 +45,20 @@
...
+    let!(:targets) { create_list(:target, 9, user: user) }
```
Comment (gonzaM) [inline spec/requests/api/v1/targets/create_spec.rb]: Can't the user have 10 targets? I think the error should be raised if it's more than 10

File: spec/requests/api/v1/targets/create_spec.rb (hunk 1 above)
Comment (gonzaM) [reply spec/requests/api/v1/targets/create_spec.rb]: For example: `'when the user already has the max number of targets'`

File: spec/requests/api/v1/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+describe 'GET api/v1/targets' do
+  let(:user) { create(:user) }
+  let!(:targets) { create_list(:target, 10, user: user) }
+
+  subject do
+    get api_v1_targets_path, headers: auth_headers, as: :json
+  end
+
+  context 'for current user' do
```
Comment (gonzaM) [inline spec/requests/api/v1/targets/index_spec.rb]: This should not be a context

File: app/validators/target_validator.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class TargetValidator < ActiveModel::Validator
```
Comment (TimoPeraza) [inline app/validators/target_validator.rb]: It's okey, but maybe is better have only in the target model:
```
validate :targets_count_limit, on: :create

def targets_count_limit
     return if self.user.targets.size < 10

     errors.add(:message, I18n.t('models.target.error.to_many_targets'))
end
```
I don't know what think the rest of the reviewers

File: app/validators/target_validator.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  def validate(record)
+    count = User.find(record[:user_id]).targets.count
+    return record.errors.add(:message, I18n.t('models.target.error.to_many_targets')) if count >= 10
```
[the remark this answers is not in the document]
Comment (amaury901130) [reply app/validators/target_validator.rb]: what's the difference?

File: spec/requests/api/v1/targets/create_spec.rb (hunk 2 above)
Comment (amaury901130) [reply spec/requests/api/v1/targets/create_spec.rb]: yes the problem I already create 1 target before at the beginning.

File: app/validators/target_validator.rb (hunk 4 above)
Comment (amaury901130) [reply app/validators/target_validator.rb]: In this way, you add more logic to the model class, In my personal opinion, it's more practical to keep separate all logic outside the model class.
also, class Validator has useful functions.

File: spec/models/target_spec.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 describe Target, type: :model do
   describe 'create validation' do
-    subject { create(:target) }
+    subject { create(:target, user: create(:user)) }
```
Comment (TimoPeraza) [inline spec/models/target_spec.rb]: maybe is better:
```
let(:user) { create(:user) }
subject { create(:target, user: user) }
```

File: app/validators/target_validator.rb (hunk 4 above)
Comment (TimoPeraza) [reply app/validators/target_validator.rb]: But now you don't have any logic in the target model, so add logic to its isn't a problem. Separate logic is good if you have many queries or filters that aren't related and the size of the model is too big. I believe this validation should be in the model but maybe i'm wrong.

File: spec/requests/api/v1/targets/create_spec.rb [hunk 7]
```diff
@@ -45,4 +45,20 @@
       expect(json[:topic][:id]).to eq(topic_id)
     end
   end
+
+  context 'when the user already has the max number of targets' do
+    let!(:targets) { create_list(:target, 9, user: user) }
+
+    it 'reponse bad request' do
```
Comment (TimoPeraza) [inline spec/requests/api/v1/targets/create_spec.rb]: maybe is better: it 'returns a bad request'

File: spec/requests/api/v1/targets/create_spec.rb [hunk 8]
```diff
@@ -45,4 +45,20 @@
...
+      subject
+
+      expect(response).to have_http_status(:bad_request)
+    end
+
+    it 'reponse error message' do
```
Comment (TimoPeraza) [inline spec/requests/api/v1/targets/create_spec.rb]: maybe is better: it 'returns an error message'

File: spec/models/target_spec.rb [hunk 9]
```diff
@@ -22,7 +22,8 @@
 
 describe Target, type: :model do
   describe 'create validation' do
-    subject { create(:target) }
+    let(:user) { create(:user) }
+    subject { create(:target, user: user) }
```
Comment (gonzaM) [inline spec/models/target_spec.rb]: I would align the lets.

File: spec/requests/api/v1/targets/create_spec.rb [hunk 10]
```diff
@@ -1,8 +1,8 @@
 require 'rails_helper'
 
 describe 'POST api/v1/targets' do
-  let(:target) { create(:target) }
-  let(:user) { target.user }
+  let(:user) { create(:user) }
+  let(:target) { create(:target, user: user) }
```
Comment (gonzaM) [inline spec/requests/api/v1/targets/create_spec.rb]: Same as above

**Pull request merged**